packages/@aws-cdk/aws-lambda-nodejs/lib/function.ts

```diff
@@ -0,0 +1,152 @@
+import * as lambda from '@aws-cdk/aws-lambda';
+import * as cdk from '@aws-cdk/core';
+import { spawnSync } from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+
+/**
+ * Properties for a NodejsFunction
+ */
+export interface NodejsFunctionProps extends lambda.FunctionOptions {
+  /**
+   * Path to the entry file (JavaScript or TypeScript).
+   *
+   * @default - `index.ts` or `index.js` in the folder named as the construct's id.
+   */
+  readonly entry?: string;
+
+  /**
+   * The name of the exported handler in the entry file.
+   *
+   * @default handler
+   */
+  readonly handler?: string;
+
+  /**
+   * The runtime environment. Only runtimes of the Node.js family are
+   * supported.
+   *
+   * @default Runtime.NODEJS_12_X
```

Comment:
Default should be deduced from `process.versions.node`

Reply:
Closest version?

something like:
```
if (>=12) {
  return 12.x
} else {
  return 10.x
}
```

Reply:
yes